### DISCUSSION TITLE
No clear way to split tabs graphically or with keybindings.

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#748] [Ideas]
### What feature would you like to see?

Currently, there is no clear graphical way to split specific tabs. Even if you use keybindings, there is no documentation or indicator as to what they are within the browser (I had to go to Reddit in order to figure them out, and it is still unclear as to how to rearrange split tabs.) a few ways to fix this would be in the right-click menu when clicking a tab to have a "select tab" option that has the keybindings written in gray on the right (how this is usually handled in the vast majority of browsers), a tutorial at instillation informing what the shortcuts are, and a graphical overview when hovering over a tab, along with once 2 or more tabs are in a workspace for an icon to appear in the header to change the layout. Any one of these options, or even just a "shortcuts"  tab on the help section of the website would be an improvement to the current situation.

### Other details

This is the main marketed function of this browser, yet it is what gives average and even knowledgeable users the most issue. 
 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/714_

**Comment by DaniilSkLi** [comment]
I faced the same difficulties. It's not intuitive, and problematic when there are many tabs and I would like to see only certain ones in this mode. As an option, you can do as in code editors, that you can just take the site icon from the tab bar, and drag it into the viewport, and that would indicate how the space will be divided depending on the cursor position, as well as perhaps how to group them in the tab bar, that would be visible that these tabs in the group. It would also be good to make it possible to change tabs in this mode, because now it is not very convenient, when for example 3 tabs, make a grid, and you get 2 small and 1 large tab, and often would like that the big one was for example used, and the small ones as secondary, and you have to leave this mode, sort them on the panel and try again, and it is not intuitive.

This is just my view of the problem) Just came across this while looking for information on this.

**Comment by mutech** [comment]
Yeah, the split view is really under-supported or over-advertised. I was searching for a way to test it and didn't find it. 

I guess I see where the problem is coming from. There are a whole lot of abstractions that overlap and are incredibly difficult to condense into a consistent user interface.

I struggled with these:

- Pinned tab vs. Bookmark (in the bar): Both serve a very similar purpose, so I don't want both. But to replace my use of bookmarks in the tab bar, I would need to rename pinned tabs.
- Tabs on the side bar: This is actually better to organize tabs, but it's inconvenient and unintuitve when browsing with tabs. I find myself moving the mouse in the top area (when typing URLs) and then need to move to the center left to manipulate the related tab. Visually, the new tab is far away from the action (the URL).
- Profiles, work spaces, containers, tabs and bookmarks are confusing when put together, and while browsing, all of them are put together (if you use them). My understanding is this: 
  - Profiles are fully isolated environments, like different browser installations. This is useful to comply with work policies, to play with dangerous plugins, or whatever the use case might be. Here the button that opens a new profile is confusing, because it's not there in profile windows (or just sometimes).
  - Work spaces: This corresponds to my concept of a "context", I'm working on project X, making a pause to watch videos, surf casually, or whatever and if I can make myself switch according to my mood, I get a well tuned UI for the respective purpose.
  - Containers: That seems to be a cross over between profiles and work spaces, in that it does some isolation like profiles and this is usually related to a usage context like work spaces. It allows me to test functionality with predefined SSO (use one MS account for a test user and another for myself). Cool. But it's hard to grasp the state of things in the UI. Certainly more convenient than to use different profiles, but still complex. But so is the use case.
  - Tabs are the culprit here, because they are subject to work spaces and containers. They are also tied to a displayed URL.

And here split tabs make it really hard, because they don't fit in the UI paradigm of the browser at all. You don't want a tab-side bar in each split view. If more than one tab is displayed, where to show the address bar? If multiple tabs are shown and the UI elements show the "current" tab, then there is no chrome to manipulate tabs.

Zen has a whole lot of potential. I played with it just a day and it feels really good. But to make it great, it needs to provide a straight forward concept for how to present these concepts or it might need to create a whole different paradigm to make accomplishing what these abstractions can do for users much easier and more intuitive.

The split tabs are probably one of the biggest challenges. But things like moving a tab from one Zen window into another and preserving the UI state might be just as difficult to get right.

I'm definitely awed by how good Zen feels already. Keep up the good work!
